src/generate/peripheral.rs

```diff
@@ -39,12 +39,23 @@ pub fn render(
         pub struct #name_pc { _marker: PhantomData<*const ()> }
 
         unsafe impl Send for #name_pc {}
+        #[cfg(feature = "ownership")]
+        unsafe impl Sync for #name_pc {}
 
         impl #name_pc {
             /// Returns a pointer to the register block
             pub fn ptr() -> *const #base::RegisterBlock {
                 #address as *const _
             }
+            /// Returns a mutable pointer to the register block
+            pub fn ptr_mut() -> *mut #base::RegisterBlock {
+                #address as *mut _
+            }
+            /// Destroys the zero sized handle in favor of a `&'static mut` reference to the
+            /// register block
+            pub fn into_static_mut(self) -> &'static mut #base::RegisterBlock {
+                unsafe { &mut*#name_pc::ptr_mut() }
```

Comment:
Is that going to compile (&mut* back to back with the name)? Anyway, I'd prefer consistency.

```suggestion
                unsafe { &mut *#name_pc::ptr_mut() }
```